File: 프로그래머스/0/181855. 문자열 묶기/문자열 묶기.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>


using namespace std;

int solution(vector<string> strArr) {
    int answer = 0;
    
    unordered_map<int,int> count;
    
    for(string s:strArr)
    {
        count[s.length()]++;
    }
    
    for(auto a:count)
    {
        answer=max(answer,a.second);
    }
    
    return answer;
}

File: 프로그래머스/0/181855. 문자열 묶기/문자열 묶기.py
def solution(strArr):
    answer = 0
    counter = {}
    
    # 각 문자열의 길이를 카운트
    for arr in strArr:
        length = len(arr)
        if length not in counter:
            counter[length] = 0
        counter[length] += 1
    
    # 가장 많은 문자열 길이의 개수 찾기
    answer = max(counter.values())
    
    return answer
